tests/accumulate-headers-after-304.js

```diff
@@ -0,0 +1,178 @@
+// Daft-based Squid Tests
+// Copyright (C) The Measurement Factory.
+// http://www.measurement-factory.com/
+
+// Tests whether an HTTP proxy does not allow to increase an HTTP response header
+// via 304 responses more than the specified limit.
+
+import HttpTestCase from "../src/test/HttpCase";
+import Field from "../src/http/Field";
+import Body from "../src/http/Body";
+import Resource from "../src/anyp/Resource";
+import * as ConfigurationGenerator from "../src/test/ConfigGen";
+import * as AddressPool from "../src/misc/AddressPool";
+import * as FuzzyTime from "../src/misc/FuzzyTime";
+import * as Gadgets from "../src/misc/Gadgets";
+import * as Config from "../src/misc/Config";
+import assert from "assert";
+import Test from "../src/overlord/Test";
+
+// TODO: make configurable
+const MaxHeaderSize = 65536;
+
+Config.Recognize([
+    {
+        option: "smp",
+        type: "Boolean",
+        default: "false",
+        description: "In this mode MISS, UPDATE and HIT requests will go to different proxy SMP workers",
+    },
+    {
+        option: "dut-cache",
+        type: "String",
+        enum: [ 'mem', 'disk', 'all' ],
+        description: "Turns on rock disk cache",
+    },
+]);
+
+export default class MyTest extends Test {
+
+    _configureDut(cfg) {
+        const memCache = Config.dutCache() === 'mem' || Config.dutCache() === 'all';
+        const diskCache = Config.dutCache() === 'disk' || Config.dutCache() === 'all';
+        cfg.memoryCaching(memCache || !Config.smp()); // always turn on memory cache in non-smp mode
+        cfg.diskCaching(diskCache && Config.smp()); // turn on rock only in smp mode
+
+        if (Config.smp()) {
+            cfg.workers(4);
+            cfg.dedicatedWorkerPorts(true);
+            this._workerListeningAddresses = cfg.workerListeningAddresses();
+        } 
+    }
+
+    static Configurators() {
+        const configGen = new ConfigurationGenerator.FlexibleConfigGen();
+
+        configGen.smp(function *(cfg) {
+            yield false;
+            yield true;
+        });
+
+        configGen.dutCache(function *(cfg) {
+            yield "mem";
+            if (cfg.smp()) {
+                yield "disk";
+                yield "all";
+            }
+        });
+        
+        return configGen.generateConfigurators();
+    }
+
+    async testOne(headerSize, updateSuccess) {
```

Comment:
There are quite a few response headers in this test. Please rename this variable to give the code reader more information. I have not checked all the details, but here is one suggestion (adjust as needed while addressing other change requests):

```suggestion
    async testOne(updatedResponseHeaderSize) {
```

Please also remove the unused second parameter.